packages/components/forms/src/text-input/TextInput.mdx

```diff
@@ -0,0 +1,78 @@
+---
+title: 'TextInput'
+type: 'component'
+status: 'stable'
+slug: /components/forms/src/text-input/
+github: 'https://github.com/contentful/forma-36/tree/master/packages/components/forms/src/text-input'
+storybook: 'https://f36-storybook.contentful.com/?path=/story/form-elements-textinput--basic'
+typescript: ./TextInput.tsx
+---
+
+`TextInput` is a form component, that allows users to input text or numbers and should be used in a form context.
+
+## Table of contents
+
+- [How to use TextInput](#how-to-use-TextInput)
+- [Component variations](#component-variations)
+- [Content recommendations](#content-recommendations)
+- [Accessibility](#accessibility)
+- [Props](#props)
+
+## How to use TextInput
+
+The TextInput component can be configured in a number of different ways. Use this guidance to determine the correct variation based on the context.
+
+TextInput is supporting different types of inputs, like `'text' | 'email' | 'file' | 'number' | 'password' | 'search' | 'url'`, where `text` is a default.
+Label property is always required and passed to the `<input />` as `aria-label`.
+
+## Code examples
+
+```jsx
+<React.Fragment>
+  <TextInput type="email" name="email" label="Email address" />
+</React.Fragment>
+```
+
+## TextInput variations
+
+### Character count (`countCharacters`)
+
+The TextInput can be configured to display a character count. This is useful when an input field has a character limit.<br/>
+**Note**: The `maxLength` property must be also provided.
+
+```jsx
+<TextInput
+  type="text"
+  label="name"
+  name="name"
+  id="input-1"
+  placeholder="Type your name"
+  countCharacters
+  maxLength={150}
+/>
+```
+
+### TextInput with copy button (`withCopyButton`)
+
+We provide a variant of the TextInput that is a composed with CopyButton. In order to use it add `withCopyButton` property to your implementation.
+
+```jsx
+<TextInput label="Token" name="token" id="input-2" withCopyButton />
+```
+
+## Content recommendations
+
+- Use direct, succinct copy that helps the user to successfully complete the form
+- Do not preface the instructions with introductory text, such as "please"
+- To add additional context, link out to documentation
+- In cases where additional context is unnecessary, help text can be substituted for text in the label
+- Do not use punctuation for labels, help text and links
```

Comment:
```suggestion
- Do not use punctuation in labels, help text and links
```

Idk honestly... It sounds better for me with "in"
but I'm not a grammar teacher so feel free to ignore this

Reply:
This is actually not valid anymore since this component does not have labels, help texts nor links

Reply:
we could pass the label as `placeholder` and `aria-label` if the placeholder isn't defined, for cases where the TextInput won't be used with the `FormControl`, WDYT?

Reply:
@massao I don't know if it would not be confusing. I would say, we enforce aria-label for this component and that's it. I think a placeholder is not required? Label is handled as a separate component. I would rename `label` to `aria-label` but I want to do it in separate PR, to not mix 2 different components (this change have to be done for BaseInput as well)

Reply:
placeholder can be required depending on the context, like in a search bar or something, it may not have the label, the github search bar itself as example, the `aria-label` and `placeholder` have the same value. `aria-label` to be accessible to screen readers and `placeholder` to be visually accessible.

Reply:
I would still keep them as separate properties - `aria-label` and `placeholder`. Placeholder is not required for input to pass a11y check, although it does not mean it’s usable. I will check with designers if they have any cases in mind then you might want to use an input without a placeholder.